elytron/WFLY-14735-Management-FORM-authentication.adoc

```diff
@@ -0,0 +1,280 @@
+which will accept an arbitrary set of attributes during authentication.  The existing security realms
+within WildFly Elytron that handle username/password based authentication will be updated to support
+this new dynamic evidence type.
+
+==== core-management
+
+The number of attributes related to security on the management interfaces has grown with different
+permutations needed for different situations.  To reduce the complexity a new resource `policy=security`
+will be added to the `core-management` subsystem, this will allow references to the sasl or http 
+authentication factories as well as to ssl-contexts to be used.  This new policy resouce will also be
+able to reference the new context made available from the `elytron-authentication` subsystem.
+
+==== Management Interfaces
+
+On the management interfaces a new attribute `use-core-management-security-policy` will be added,
+this will be a boolean and when set to `true` the configuration defined on the `policy=security`
+resource will be used instead of the individual attributes on the management interface.  This attribute
+will default to `false` and will need to be explicitly configured.
+
+The existing attributes, `ssl-context`, `sasl-authentication-factory`, and 
+`http-authentication-factory` will not be mutually exclusive / `altertives` to the new attribute,
+instead individual capabilities can override the capabilities from the security policy.
+
+==== Out Of Scope
+
+The RFE will not be making changes to the SASL authentication for native access, the out of the
+box configuration will still default to supporting the JBOSS_LOCAL_USER and DIGEST SASL mechanisms.
+
+This RFE will also not be changing the getting started experience adding the initial users to the
+server using the add-user utility and this RFE will not be changing the properties files used to
+contain the initial users.
+
+==== Existing Configuration
+
+This RFE only adds a new capability that can be used to provide an alternative approach to securing
+HTTP access to the management interfaces, existing configurations using Elytron capabiltiies 
+directly to secure the management interfaces will continue to work as they always did.  There are no
+current plans to disable securing the management interfaces as they have been secured previously.
+
+==== Default Configuration
+
+The new subsystem will be present in the default configuration with a FORM configuration to make
+use of the existing `ManagementDomain` for authentication.
+
+The management interfaces however will not use FORM authentication out of the box.  The reason is 
+that when activating FORM authentication TLS should also be enabled for the management interfaces
+at the same time, this way we can document the steps and considerations to take when activating
+this feature.
+
+=== Nice-to-Have Requirements
+
+The CLI will have a new security command adding to activate FORM authentication for the HTTP management
```

Comment:
@jfdenise FYI this RFE may have a nice to have requirement of another command in the CLI to talk about at some point.

Reply:
@darranl , should we log another RFE or you plan to have it included in this RFE?

Reply:
@jfdenise I think for now as it in the nice to have sesction keep in this RFE - if we don't get to it then split it out to add later.  Otherwise I think it has more issues getting the RFE through.